src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCheckedParameterTests.cs

```diff
@@ -62,65 +64,44 @@ public class C
             var compilation = CompileAndVerify(source, options: TestOptions.ReleaseExe);
             compilation.VerifyIL("C.M", @"
 {
-  // Code size       10 (0xa)
-  .maxstack  1
- ~IL_0000:  ldarg.2
-  IL_0001:  brtrue.s   IL_0009
-  IL_0003:  newobj     ""System.Exception..ctor()""
-  IL_0008:  throw
- -IL_0009:  ret
+    // Code size       15 (0xf)
+    .maxstack  1
+    ~IL_0000:  ldarg.2
+    IL_0001:  brtrue.s   IL_000e
+    IL_0003:  ldstr      ""y""
+    IL_0008:  newobj     ""System.ArgumentNullException..ctor(string)""
+    IL_000d:  throw
+    -IL_000e:  ret
 }", sequencePoints: "C.M");
             // Debug
             compilation = CompileAndVerify(source, options: TestOptions.DebugExe);
             compilation.VerifyIL("C.M", @"
 {
-      // Code size       11 (0xb)
-      .maxstack  1
-     ~IL_0000:  ldarg.2
-      IL_0001:  brtrue.s   IL_0009
-      IL_0003:  newobj     ""System.Exception..ctor()""
-      IL_0008:  throw
-     -IL_0009:  nop
-     -IL_000a:  ret
+    // Code size       16 (0x10)
+    .maxstack  1
+    ~IL_0000:  ldarg.2
+    IL_0001:  brtrue.s   IL_000e
+    IL_0003:  ldstr      ""y""
+    IL_0008:  newobj     ""System.ArgumentNullException..ctor(string)""
+    IL_000d:  throw
+    -IL_000e:  nop
+    -IL_000f:  ret
 }", sequencePoints: "C.M");
         }
 
         [Fact]
         public void TestNullCheckedParamWithOptionalNullParameter()
         {
-            // PROTOTYPE : Should give warning that the default value is null on a null-checked parameter.
             var source = @"
 class C
 {
     public static void Main() { }
     void M(string name! = null) { }
 }";
-
-            // Release
-            var compilation = CompileAndVerify(source, options: TestOptions.ReleaseExe);
-            compilation.VerifyIL("C.M", @"
-{
-    // Code size       10 (0xa)
-    .maxstack  1
-   ~IL_0000:  ldarg.1
-    IL_0001:  brtrue.s   IL_0009
-    IL_0003:  newobj     ""System.Exception..ctor()""
-    IL_0008:  throw
-   -IL_0009:  ret
-}", sequencePoints: "C.M");
-            // Debug
-            compilation = CompileAndVerify(source, options: TestOptions.DebugExe);
-            compilation.VerifyIL("C.M", @"
- {
-    // Code size       11 (0xb)
-    .maxstack  1
-   ~IL_0000:  ldarg.1
-    IL_0001:  brtrue.s   IL_0009
-    IL_0003:  newobj     ""System.Exception..ctor()""
-    IL_0008:  throw
-   -IL_0009:  nop
-   -IL_000a:  ret
-}", sequencePoints: "C.M");
+            CreateCompilation(source).VerifyEmitDiagnostics(
```

Comment:
>VerifyEmitDiagnostics [](start = 38, length = 21)

Can we use `VerifyDiagnostics()` rather than `VerifyEmitDiagnostics()` for these tests?

Reply:
Consider moving the tests which aren't meant to exercise lowering or emit code to Semantic/Semantics/NullCheckedParameterTests.cs

---
In reply to: [303094599](https://github.com/dotnet/roslyn/pull/36979#discussion_r303094599) [](ancestors = 303094599)